packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -126,6 +127,41 @@ class CupertinoCheckbox extends StatefulWidget {
   /// Defaults to [CupertinoColors.activeBlue].
   final Color? activeColor;
 
+  /// {@template flutter.material.checkbox.fillColor}
+  /// The color used to fill this checkbox.
+  ///
+  /// Resolves in the following states:
+  ///  * [WidgetState.selected].
+  ///  * [WidgetState.hovered].
+  ///  * [WidgetState.focused].
+  ///  * [WidgetState.disabled].
+  ///
+  /// {@tool snippet}
+  /// This example resolves the [fillColor] based on the current [WidgetState]
+  /// of the [CupertinoCheckbox], providing a different [Color] when it is
+  /// [WidgetState.disabled].
+  ///
+  /// ```dart
+  /// CupertinoCheckbox(
+  ///   value: true,
+  ///   onChanged: (_){},
+  ///   fillColor: WidgetStateProperty.resolveWith<Color>((Set<WidgetState> states) {
+  ///     if (states.contains(WidgetState.disabled)) {
+  ///       return Colors.orange.withOpacity(.32);
+  ///     }
+  ///     return Colors.orange;
+  ///   })
+  /// )
+  /// ```
+  /// {@end-tool}
+  /// {@endtemplate}
+  ///
+  /// If null, then the value of [activeColor] is used in the selected
+  /// state. If that is also null, then [CupertinoColors.inactiveGray] is used in
+  /// the disabled state, [CupertinoColors.activeBlue] is used in the
```

Comment:
```suggestion
  /// the disabled state, and [CupertinoColors.activeBlue] is used in the
```

Reply:
It's confusing that `activeColor` above also resolves to `CupertinoColors.activeBlue`. What parameters are  prioritized when it comes to setting the color of the checkbox?

Reply:
If any one is giving null, it resolves to bottom one
1. fillColor
2. activeColor
3. CupertinoColors.activeBlue

Reply:
Yes, but that is not clear to someone reading the documentation. The documentation for `fillColor` and `activeColor` should both clearly specify this order of precedence.

I'd recommend appending this to `activeColor`'s docs:

```dart
  /// If [fillColor] returns a non-null color in the [WidgetState.selected]
  /// state, it will be used instead of this color.
  ```

Reply:
Then, since `activeColor` already defaults to `CupertinoColors.activeBlue`, `fillColor` only needs to default to `activeColor` in the selected state, so that there is a single source of truth for what the final default color is.

Reply:
I have updated code to make it simple and in sync with docs. Please review and let me know if it makes more sense now.